kratos/modeler/clean_up_problematic_triangles_modeler.h

```diff
@@ -0,0 +1,337 @@
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    /**
+     * @brief Get the nodes of the model part.
+     * @param rThisModelPart Reference to the model part.
+     * @return The nodes of the model part.
+     * @tparam TEntityType The entity type to get from the model part. Can be Element,  Condition or Geometry.
+     */
+    template <typename TEntityType>
+    static void RemoveEntitiesAndNodes(ModelPart& rThisModelPart);
+
+    /**
+     * @brief Clean up the problematic geometries (null area geometries) in the mesh.
+     * @details For the moment it will be assumed that the problematic geometries are triangles with two nodes in the same position. This means several this:
+     * - The triangle is degenerated, one of the nodes is redundant and should be cleaned up.
+     * - The triangle must be removed as well. The sides must be reconnected, so the mesh is water-tight.
+     * - The nodes and entities must be renumbered
+     * @param rThisModelPart Reference to the model part to clean up into.
+     * @param rEntityType The entity type to create in the model part. Can be "element", "condition" or "geometry".
+     * @param FirstNodeId The first node ID to create in the model part.
+     * @param FirstEntityId The first entity ID to create in the model part.
+     * @param AreaTolerance The tolerance to consider a geometry as problematic.
+     * @tparam TEntityType The entity type to create in the model part. Can be Element, Condition or Geometry.
+     */
+    template <typename TEntityType>
+    static void CleanUpProblematicGeometries(
+        ModelPart& rThisModelPart,
+        const IndexType FirstNodeId = 1,
+        const IndexType FirstEntityId = 1,
+        const double AreaTolerance = 1.0e-4
+        );
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    ///@}
+
+}; // Class CleanUpProblematicTrianglesModeler
+
+// Helper function definitions specialization
+template <>
+void CleanUpProblematicTrianglesModeler::RemoveEntitiesAndNodes<Element>(ModelPart& rThisModelPart);
+
+template <>
+void CleanUpProblematicTrianglesModeler::RemoveEntitiesAndNodes<Condition>(ModelPart& rThisModelPart);
+
+///@}
+
+///@name Type Definitions
+///@{
+
+///@}
+///@name Input and output
+///@{
+
+/// input stream function
+inline std::istream & operator >>(std::istream& rIStream,
+                                  CleanUpProblematicTrianglesModeler& rThis);
```

Comment:
```suggestion
inline std::istream & operator >>(
    std::istream& rIStream,
    CleanUpProblematicTrianglesModeler& rThis);
```
Minor style guide comment.